src/controllers/Controller.js

```diff
@@ -302,6 +306,77 @@ async loginRestaurant(req, res) {
     }
   }
 
+  async servePaymentForm(req, res) {
+    try {
+      const filePath = './public/payment-form.html';
+      fs.readFile(filePath, (err, data) => {
+        if (err) {
+          res.writeHead(500, { 'Content-Type': 'text/plain' });
+          res.end('Failed to fetch payment-form.html ' + ' ' + err);
+        } else {
+          res.writeHead(200, { 'Content-Type': 'text/html' });
+          res.end(data);
+        }
+      });
+    } catch (error) {
+      response(res, { status: 400, data: error.message });
+    }
+  }
+
+  async getClientPaymentToken(req, res) {
+    try {
+      const filePath = './public/scripts/get-client-token.js';
+      fs.readFile(filePath, (err, data) => {
+        if (err) {
+          res.writeHead(500, { 'Content-Type': 'text/plain' });
+          res.end('Failed to fetch get-client-token.js ' + ' ' + err);
+        } else {
+          res.writeHead(200, { 'Content-Type': 'text/html' });
+          res.end(data);
+        }
+      });
+    } catch (error) {
+      response(res, { status: 400, data: error.message });
+    }
+  }
+
+  async submitPayment(req, res) {
+    try {
+      let data = '';
+
+    req.on('data', (chunk) => {
```

Comment:
I think you can directly fetch the body by doing req.body and use your two keys `token` and `amount`